src/Mod/Sketcher/Gui/CommandConstraints.cpp

```diff
@@ -840,16 +841,15 @@ class DrawSketchHandlerGenConstraint: public DrawSketchHandler
 
     virtual void activated(ViewProviderSketch *)
     {
-        Gui::Selection().rmvSelectionGate();
+        ongoingSequences = new std::set<int>(); //TODO: make it contain more values
+        _tempOnSequences = new std::set<int>();
+        selFilterGate = new GenericConstraintSelection(sketchgui->getObject());
 
-        // Estimate allowed selections from the first types in allowedSelTypes
-        int allowedSelTypes = 0;
-        for (std::vector< std::vector< SelType > >::const_iterator it = cmd->allowedSelSequences.begin();
-             it != cmd->allowedSelSequences.end(); ++it) {
-            allowedSelTypes = allowedSelTypes | *((*it).begin());
-        }
-        GenericConstraintSelection* selFilterGate = new GenericConstraintSelection(sketchgui->getObject());
-        selFilterGate->setAllowedSelTypes(allowedSelTypes);
+        resetOngoingSequences();
+
+        selSeq = *(new std::vector<SelIdPair>());
```

Comment:
> I guess you are mixing up C++ with JAVA.
Indeed I was! Never knew such initiations are handled automatically. Is that what you meant by "create on stack" in the other comment?